include/onnxruntime/core/framework/sparse_tensor.h

```diff
@@ -1,72 +1,427 @@
+ * 
  */
 
 class SparseTensor final {
  public:
+  /// <summary>
+  /// This constructs an instances that points to user defined buffers.
+  /// Make use of Use* functions to supplied format specific indices that
+  /// reside in the user supplied buffers. Format specific indices are expected
+  /// to be supplied via Use*() methods.
+  /// </summary>
+  /// <param name="elt_type">MlDataType</param>
+  /// <param name="dense_shape">a shape of original tensor in dense form</param>
+  /// <param name="values_shape">shape for user supplied values</param>
+  /// <param name="values_data">a pointer to values</param>
+  /// <param name="location">description of the user allocated memory</param>
   SparseTensor(MLDataType elt_type,
-               const TensorShape& shape,
-               size_t nnz,
+               const TensorShape& dense_shape,
+               const TensorShape& values_shape,
                void* values_data,
-               void* indices_data,
-               const OrtMemoryInfo& memory_info);
+               const OrtMemoryInfo& location);
 
+  /// <summary>
+  /// A constructor that would memorize allocation details but
+  /// will do nothing. All the data would come via Make*() interfaces.
+  /// </summary>
+  /// <param name="elt_type"></param>
+  /// <param name="dense_shape"></param>
+  /// <param name="allocator"></param>
   SparseTensor(MLDataType elt_type,
-               const TensorShape& shape,
-               size_t nnz,
+               const TensorShape& dense_shape,
                std::shared_ptr<IAllocator> allocator);
 
-  ~SparseTensor() = default;
+  SparseTensor() noexcept;
+
+  ~SparseTensor();
 
   // For now, disallow all copy, assignment, and move.
-  ORT_DISALLOW_COPY_ASSIGNMENT_AND_MOVE(SparseTensor);
+  ORT_DISALLOW_COPY_AND_ASSIGNMENT(SparseTensor);
 
   // Returns the number of entries in the values tensor (aka "NNZ" or "number of nonzero values")
+  // For block sparse formats this may include some zeros in the blocks are considered non-zero.
   size_t NumValues() const { return static_cast<size_t>(values_.Shape().Size()); }
 
-  const Tensor& Values() const {
+  /// <summary>
+  /// Read only accessor to values
+  /// </summary>
+  /// <returns></returns>
+  const Tensor& Values() const noexcept {
     return values_;
   }
 
-  const Tensor& Indices() const {
-    return indices_;
+  SparseTensor(SparseTensor&& o) noexcept;
+  SparseTensor& operator=(SparseTensor&& o) noexcept;
+
+  /// <summary>
+  /// Returns SparseFormat that the instance currently holds
+  /// if the value returned in kUndefined, the instance is not populated
+  /// </summary>
+  /// <returns></returns>
+  SparseFormat Format() const noexcept {
+    return format_;
   }
 
-  const TensorShape& Shape() const {
-    return shape_;
+  /// <summary>
+  /// Returns a would be dense_shape
+  /// This does not describe shapes of values of indices or values.
+  /// </summary>
```

Comment:
nit: this clarification wouldn't be necessary if the method was called something like DenseShape()